index.php

```diff
@@ -145,93 +145,94 @@
             </div>
         </div>
 
+<?php
+include 'connectdb.php';
+$dishesPerPage = 12;
+$page = isset($_GET['page']) ? (int)$_GET['page'] : 1;
+$offset = ($page - 1) * $dishesPerPage;
+
+$filters = array();
+$params = array();
+
+// Always add the note filter for "r"
+$filters[] = "note = :note";
+$params[':note'] = 'r';
+
+// Handle search keyword
+if (isset($_GET['keyword']) && !empty($_GET['keyword'])) {
+    $filters[] = "dish_name LIKE :keyword OR description LIKE :keyword";
+    $params[':keyword'] = '%' . $_GET['keyword'] . '%';
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider sanitizing SQL LIKE input</b></div>
<div id="fix">

The SQL LIKE query with `%keyword%` pattern could benefit from input sanitization to prevent SQL injection and improve performance. Consider using prepared statements.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
    $sanitized_keyword = preg_replace('/[%_]/', '\\$0', $_GET['keyword']);
    $filters[] = "dish_name LIKE :keyword OR description LIKE :keyword";
    $params[':keyword'] = '%' . $sanitized_keyword . '%';
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/6#issuecomment-2543172057>#ef655a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged